lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -4560,6 +4560,12 @@ bool LLVMToSPIRVBase::transExecutionMode() {
           break;
         AddSingleArgExecutionMode(static_cast<ExecutionMode>(EMode));
       } break;
+      case spv::internal::ExecutionModeNamedBarrierCountINTEL: {
+        unsigned NBarrierCnt;
+        N.get(NBarrierCnt);
+        BF->addExecutionMode(new SPIRVExecutionMode(
+            BF, static_cast<ExecutionMode>(EMode), NBarrierCnt));
+      } break;
```

Comment:
Here should be a check, if the appropriate extension is enabled.
Also we need to add the appropriate capability.